src/sentry/integrations/github_enterprise/repository.py

```diff
@@ -14,6 +15,20 @@ class GitHubEnterpriseRepositoryProvider(GitHubRepositoryProvider):
     logger = logging.getLogger('sentry.plugins.github_enterprise')
     repo_provider = 'github_enterprise'
 
+    def _validate_repo(self, client, installation, repo):
+        try:
+            repo_data = client.get_repo(repo)
+        except Exception as e:
+            installation.raise_error(e)
+
+        try:
+            # make sure installation has access to this specific repo
+            client.get_commits(repo)
+        except ApiError as e:
+            raise IntegrationError('You must grant Sentry access to {}'.format(repo))
```

Comment:
Do repository names contain non-ascii bytes? You could have a UnicodeError here.

Reply:
I don't _think_ they let you? It seems like when I try to create a new repo they replace non-ascii characters with "-"

Reply:
Could you make this a unicode string anyway? There's really no downside to it.